modules/ROOT/pages/java/java-invoke-method.adoc

```diff
@@ -54,9 +76,275 @@ You can invoke static Java methods like this:
+* Use the Invoke operation in the Java module to invoke the `hello()` and
+`add(3,4)` instance methods.
+* Run the Mule app to execute New and Invoke operations in Studio.
+
+Assume that you want to invoke methods in two simple Java classes, for example:
+
+[[hello_class]]
+.Hello Class with an Instance Method
+[source,java,linenums]
+----
+package com.examples;
+
+public class Hello {
+
+  //constructor
+  public Hello() { }
+
+  public String hello() {
+    return "helloWorld";
+  }
+}
+----
+
+[[add_class]]
+.Add Class with a Parameterized Instance Method
+[source,java,linenums]
+----
+public class Add {
+
+  public int x = 0;
+  public int y = 0;
+
+  //constructor
+  public Add(int numA, int numB) {
+	        x = numA;
+	        y = numB;
+  }
+
+  public int add(int x, int y) {
+    return x + y;
+  }
+}
+----
+
+To create a Mule app that invokes the `hello()` and `add()`:
+
+. Create a new project in Studio with a name such as `javaexamples`:
++
+In Studio, you simply select *File* -> *New* -> *Project*, provide a project
+name (`javaexamples`), and then click *Finish*.
++
+. Create a Java package for your classes:
+.. Right-click your Mule project's `src/main/java` directory from
+the *Package Explorer* tab (located on the left).
+.. Select *New* -> *Package*.
+.. Provide the package name `com.examples` in the *Name* field.
+.. Click *Finish*.
++
+Make sure that the package `com.examples` appears under the `src/main/java`
+directory.
++
+. Add the Java code for the `Hello` and `Add` classes to your new `com.examples`
+package:
+.. Right-click your new `com.examples` package in Studio, and select *New* -> *Class*.
+.. Type the *Name* `Hello.java` for the `Hello` class, and click *Finish*.
+.. Copy and paste the `Hello` class content into the `Hello.java` file.
++
+The entire file looks like this, including the `package.com.examples` at the top:
++
+[source,java,linenums]
+----
+package com.examples;
+
+public class Hello {
+
+  //constructor
+  public Hello() { }
+
+  public String hello() {
+    return "helloWorld";
+  }
+}
+----
++
+.. Right-click your new `com.examples` package in Studio, and then select
+*New* -> *Class*.
+.. Type the *Name* `Add.java`.
+.. Click *Finish*.
+.. Copy and paste the class content into the `Add.java` file.
++
+The entire file looks like this, including the `package.com.examples` at the top:
++
+[source,java,linenums]
+----
+package com.examples;
+
+public class Add {
+
+  public int x = 0;
+  public int y = 0;
+
+  //constructor
+  public Add(int numA, int numB) {
+          x = numA;
+          y = numB;
+  }
+
+  public int add(int x, int y) {
+    return x + y;
+  }
+}
+----
++
+. Click the *javaexamples* tab in Studio to return to your Mule app, and set
+up a flow for `Hello` class and `hello()` method:
+.. In *javaexamples*, provide a trigger for a flow by dragging a Scheduler
+component into the Studio canvas.
++
+Optional: You can also set the Scheduler frequency to something less frequent
```

Comment:
Maybe mention what the default is.

Reply:
The Scheduler isn't that relevant to this exercise. It's just there so they can run the app. The Java module operations are the focus.